### PR TITLE
Generic Nodes

### PR DESCRIPTION
**Pull request by tnrz135**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NonCollidingNodeSystemAlgorithm/AStar.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            }
+
+            addToOpen(nodesCanGoTo, current, closed, parents);
+
+        }
+        return null;
+    }
+
+    public static void main(String[] args) {
+        //LinkedList<SystemsPos> a = getPath(ARM_LOWWW, ARM_HIGH);
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NonCollidingNodeSystemAlgorithm/AStar.java:92]: Commented out code doesn't do anything

Also the entire `main` method here is out of place

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+import java.util.HashMap;
+import java.util.LinkedList;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.Constants.systemName1;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.Constants.systemName2;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemFunctions.getAllSystemsPositions;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemFunctions.getNode;
+
+public class BlaBla {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java:15]: What does this name mean?

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBlaVertex.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
+package edu.greenblitz.tobyDetermined.Nodesssss.Bla3;
+
+import java.util.LinkedList;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.Constants.*;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeSystemFunctions.*;
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.SystemsPos;
+
+public class BlaBlaVertex {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBlaVertex.java:9]: Try finding a better name

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    public static LinkedList<SystemsPos> printAndReturnFinalPath(SystemsPos start, SystemsPos end, SystemsPos system2Pos, SystemsPos system3Pos) {
+        Pair<LinkedList<SystemsPos>, Double> finalPathAndCost = findPath(start, end, system2Pos, system3Pos);
+        printPath(finalPathAndCost.getFirst());
+        System.out.println("cost: "+finalPathAndCost.getSecond());
+        return finalPathAndCost.getFirst();
+    }
+
+    public static void main(String[] args) {
+        LinkedList<SystemsPos> a = printAndReturnFinalPath(SystemsPos.CLIMBING_ROMY, SystemsPos.CLIMBING_NOAM, SystemsPos.ARM_HIGH, SystemsPos.GRIPER_CLOSE);
+    }
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java:236]: Why is the main function here?

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+            }
+            addNeighborsToOpen(current, closedVer, openVer, currentVer, parents);
+        }
+        return null;
+    }
+
+    public static LinkedList<SystemsPos> printAndReturnFinalPath(SystemsPos start, SystemsPos end, SystemsPos system2Pos, SystemsPos system3Pos) {
+        Pair<LinkedList<SystemsPos>, Double> finalPathAndCost = findPath(start, end, system2Pos, system3Pos);
+        printPath(finalPathAndCost.getFirst());
+        System.out.println("cost: "+finalPathAndCost.getSecond());
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/Bla3/BlaBla.java:230]: Debug prints should not be in "production" code

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package edu.greenblitz.tobyDetermined.Nodesssss.TheSystemsNodes;
+
+import edu.greenblitz.tobyDetermined.Nodesssss.GBNode;
+
+public class GriperNode extends GBNode {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java:5]: Typo?

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NonCollidingNodeSystemAlgorithm/AStar.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static void printPath(LinkedList<SystemsPos> pathList) {
+        for (SystemsPos systemsPos : pathList) {
+            System.out.print(systemsPos + ", ");
+        }
+        System.out.println();
+    }
+
+    public static void addToOpen(LinkedList<SystemsPos> nodesCanGoTo, SystemsPos current, LinkedList<SystemsPos> closed, Map<SystemsPos, SystemsPos> parents) {
+       // for (SystemsPos neighbor : NodeBase.getNode(current).getNeighbors()) {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NonCollidingNodeSystemAlgorithm/AStar.java:61]: Commented out code

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        static public String systemName2 = "GRIPER";
+        static public String systemName3 = "CLIMBING";
+        static public SystemsPos system1StartingNode = ARM_ZG;
+        static public SystemsPos system2StartingNode = GRIPER_CLOSE;
+        static public SystemsPos system3StartingNode = CLIMBING_ROMY;
+    }
+
+    public static class important {
+        NodeArm a = new NodeArm(0, 'r' + 'o' + 'm' + 'y' + ' ' + 'i' + 's' + ' ' + 'a' + 'u' + 't' + 'i' + 's' + 't' + 'i' + 'c');
+    }
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:114]: Used for debug?

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+import edu.greenblitz.tobyDetermined.Nodesssss.TheSystemsNodes.NodeArm;
+import edu.wpi.first.math.Pair;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+
+import static edu.greenblitz.tobyDetermined.Nodesssss.NodeBase.SystemsPos.*;
+
+public class NodeBase {
+    public enum SystemsPos {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:14]: If `Pos` is short for `Position`, please use the full word

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+        GRIPER_CLOSE,
+        GRIPER_OPEN,
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:16]: Typo

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+        CLIMBING_ROMY,
+        CLIMBING_NOAM,
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:19]: Non-indicative names

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+
+        ARM_LOW,
+        ARM_MID,
+        ARM_HIGH,
+        ARM_ZG,
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:24]: ZG is the zeig hail position?

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+        MID_NODE;
+
+    }
+
+    public static class CreateNodes {
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:30]: I know there was some kind of Git messup so I won't be too strict on this but the initialization done here is out of scope for a PR called "**Generic** Nodes"

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+            nodeMap.get(CLIMBING_NOAM).setOtherSystemMustBeToEnter2(new SystemsPos[]{ARM_ZG});
+            nodeMap.get(CLIMBING_ROMY).setOtherSystemMustBeToEnter3(new SystemsPos[]{GRIPER_OPEN});
+            nodeMap.get(CLIMBING_NOAM).setOtherSystemMustBeToEnter3(new SystemsPos[]{GRIPER_OPEN});
+
+            nodeMap.get(ARM_LOW).setOtherSystemMustBeToEnter2(new SystemsPos[]{GRIPER_OPEN});
+            nodeMap.get(ARM_LOW).setOtherSystemMustBeToOut2(new SystemsPos[]{GRIPER_OPEN});
+            nodeMap.get(ARM_LOW).setOtherSystemMustBeToEnter3(new SystemsPos[]{CLIMBING_ROMY});
+            nodeMap.get(ARM_LOW).setOtherSystemMustBeToOut3(new SystemsPos[]{CLIMBING_ROMY});
+
+
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java:84]: Ctrl+Alt+L please :)

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java (hunk 6 above)
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java:5]: Also, could you elaborate on the reason behind creating this (and `ClimbingNode`)?

You're not adding any new logic or overriding stuff here

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/NodeArm.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
+package edu.greenblitz.tobyDetermined.Nodesssss.TheSystemsNodes;
+
+import edu.greenblitz.tobyDetermined.Nodesssss.GBNode;
+import edu.greenblitz.tobyDetermined.subsystems.telescopicArm.Claw.ClawState;
+import edu.wpi.first.math.util.Units;
+
+public class NodeArm extends GBNode {
+
+    private final static double TOLERANCE_ANGLE = Units.degreesToRadians(3);
+    private final static double TOLERANCE_LENGTH = 0.04;
```
Comment (Stormageddon37) [inline src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/NodeArm.java:10]: Seems like something to put in config?

**Review by Stormageddon37 — CHANGES_REQUESTED** [review]
Please rename "Pos" to "Position" EVERYWHERE, long variable names are nothing to be afraid of

Also please remove "bla" from your code

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java (hunk 8 above)
Comment (YHerm) [reply src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/NodeBase.java]: yes this is very important this is why its in important, to make it clear by informative name 💩.

File: src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java (hunk 6 above)
Comment (YHerm) [reply src/main/java/edu/greenblitz/tobyDetermined/Nodesssss/TheSystemsNodes/GriperNode.java]: on what?

**Pull request closed without merging**